code/controllers/subsystem/ruin_generator.dm

```diff
@@ -0,0 +1,37 @@
+SUBSYSTEM_DEF(ruin_generator)
+	name = "Ruin Generator"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_RUIN_GENERATOR
+	//The decorators that we can apply
+	var/list/datum/ruin_decorator/decorators = list()
+	//The ruin that we are waiting to gobble up
+	var/datum/map_generator/space_ruin/unused_ruin
+
+/datum/controller/subsystem/ruin_generator/Initialize(start_timeofday)
+	. = ..()
+	//Build decorator list
+	for (var/subtype in subtypesof(/datum/ruin_decorator))
+		var/datum/ruin_decorator/decorator = new subtype
+		decorators[decorator] = decorator.decorator_weight
+	//Generate the initial ruin
+	generate_ruin()
+
+/datum/controller/subsystem/ruin_generator/proc/get_ruin()
+	. = unused_ruin
```

Comment:
does that mean this will always return implicit null?
If so, why?

Reply:
The reference to the unused ruin gets stored in `.`.
`.` contains a reference to the ruin
`unused_ruin` contains a reference to the ruin.
Unused_ruin is set to null, meaning that `unused_ruin` contains a null pointer, but `.` still contains a reference to the ruin.
A new ruin will be generated, and the ruin that was consumed will be returned.